Jeremy likness minapi overview (#27724)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/minimal-apis/overview.md

---
title: Minimal APIs overview
author: JeremyLikness
description: An introduction to the fastest and easiest way to create web API endpoints with ASP.NET Core.
ms.author: jeliknes
monikerRange: '>= aspnetcore-6.0'
ms.date: 11/22/2022
uid: fundamentals/minimal-apis/overview
---

# Minimal APIs overview

> Minimal APIs are both a technology and an approach to quickly and easily create web-based APIs with HTTP endpoints in ASP.NET Core.

Reviewer: Can we rephrase this to : "Minimal APIs are a ASPNET Core micro-framework for quickly and easily creating HTTP APIs with .NET" 

or "Minimal APIs are a simplified approach for building fast HTTP APIs with ASPNET Core". 

A few things:
- Can we avoid "Web-based"? 
- Can we make sure the "Minimalistic approach" and "perf=fast" show up in the wording ?

Reviewer 2: > Can we rephrase this to : "Minimal APIs are a ASPNET Core micro-framework for quickly and easily creating HTTP APIs with .NET"

@JeremyLikness  I'll suggest a change with this and other changes.

You can build fully functioning REST endpoints with minimal code and configuration. Skip traditional scaffolding and avoid unnecessary controllers by fluently declaring
your API routes and actions. For example, the following code creates an API at the root of your web application that returns the text, "Hello World!"

```csharp
var app = WebApplication.Create(args);

app.MapGet("/", () => "Hello World!");

app.Run();
```

Of course, most APIs accept parameters as part of the route.

```csharp 
var builder = WebApplication.CreateBuilder(args);

var app = builder.Build();

app.MapGet("/users/{userId}/books/{bookId}", 
    (int userId, int bookId) => $"The user id is {userId} and book id is {bookId}");

app.Run();
```

> That's all it takes to get started, but it's not all that's available. Minimal APIs support the configuration and customization you need to scale to multiple APIs, handle complex routes, apply authorization rules and control the shape and content of your API responses. To learn more, use the table of contents navigation to jump to the topic of your choice. A good place to get started is <xref:fundamentals/minimal-apis/webapplication>.

Reviewer 3: I would move the [Differences between minimal APIs and APIs with controllers](https://learn.microsoft.com/en-us/aspnet/core/tutorials/min-web-api?view=aspnetcore-7.0&tabs=visual-studio#differences-between-minimal-apis-and-apis-with-controllers) section from the tutorial to here.

Consider offering the tutorial as "A good place to get started" rather than the cheat sheet.

```suggestion
That's all it takes to get started, but it's not all that's available. Minimal APIs support the configuration and customization you need to scale to multiple APIs, handle complex routes, apply authorization rules, and control the content of your API responses. A good place to get started is <xref:tutorials/min-web-api>.
```

Reviewer 2: slight tweak to @tdykstra 
shorter is always better per MS style guide

```suggestion
That's all it takes to get started, but it's not all that's available. Minimal APIs support the configuration and customization needed to scale to multiple APIs, handle complex routes, apply authorization rules, and control the content of API responses. A good place to get started is <xref:tutorials/min-web-api>.
```
